Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs

```diff
@@ -589,28 +582,89 @@ public bool PlayerInsertHeldEntity(EntityUid uid, EntityUid player, StorageCompo
             return false;
         }
 
-        return PlayerInsertEntityInWorld(uid, player, toInsert.Value, storageComp);
+        return PlayerInsertEntityInWorld((uid, storageComp), player, toInsert.Value);
     }
 
     /// <summary>
     ///     Inserts an Entity (<paramref name="toInsert"/>) in the world into storage, informing <paramref name="player"/> if it fails.
-    ///     <paramref name="toInsert"/> is *NOT* held, see <see cref="PlayerInsertHeldEntity(Robust.Shared.GameObjects.EntityUid)"/>.
+    ///     <paramref name="toInsert"/> is *NOT* held, see <see cref="PlayerInsertHeldEntity(EntityUid,EntityUid,StorageComponent)"/>.
     /// </summary>
+    /// <param name="uid"></param>
     /// <param name="player">The player to insert an entity with</param>
+    /// <param name="toInsert"></param>
     /// <returns>true if inserted, false otherwise</returns>
-    public bool PlayerInsertEntityInWorld(EntityUid uid, EntityUid player, EntityUid toInsert, StorageComponent? storageComp = null)
+    public bool PlayerInsertEntityInWorld(Entity<StorageComponent?> uid, EntityUid player, EntityUid toInsert)
     {
-        if (!Resolve(uid, ref storageComp) || !_sharedInteractionSystem.InRangeUnobstructed(player, uid))
+        if (!Resolve(uid, ref uid.Comp) || !_sharedInteractionSystem.InRangeUnobstructed(player, uid))
```

Comment:
Do we note have a Resolve(ref uid) yet? Probably easier to just add it.

Reply:
its missing also it would have to account for taking in Entity<XComponent> instead of Entity<XComponent?>